frontend/app/src/androidTest/java/com/fiveis/xend/integration/NavigationIntegrationTest.kt

```
@@ -50,11 +50,20 @@ class NavigationIntegrationTest {
     fun main_activity_launches_successfully() {
-        val scenario = ActivityScenario.launch(MainActivity::class.java)
-        scenario.onActivity { activity ->
-            assertNotNull(activity)
+        try {
+            val scenario = ActivityScenario.launch(MainActivity::class.java)
+            Thread.sleep(500) // Give time for initialization
+            scenario.onActivity { activity ->
+                assertNotNull(activity)
+            }
+            scenario.close()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)